### ISSUE TITLE
[Bug]: TeamsAttachmentDownloader cannot download file attachments and audios.

### ISSUE THREAD

**Issue by XiaofuHuang** [microsoft/teams-ai#1202]
### Language

Javascript/Typescript

### Version

latest

### Description

There are a lot of kinds of attachments in Teams. For example:
1. Inline image: copy and paste directly into the send box
   ![image](https://github.com/microsoft/teams-ai/assets/49138419/3fbecc63-cf8a-4653-846b-d221cf36584a)
2. Attachment Files: upload via the "Upload from this device" menu or through drag-and-drop actions. 
   ![image](https://github.com/microsoft/teams-ai/assets/49138419/102f57f6-b947-4ec9-8e9d-f1b5a426b6c5)
3. Voice: Recorded on Teams mobile by pressing a microphone icon button.
  ![image](https://github.com/microsoft/teams-ai/assets/49138419/2aa2d63a-dd8c-413e-8d66-f05ce50b4466)
......

Currently, `TeamsAttachmentDownloader` in `teams-ai` SDK only supports the download of inline images. Can it be extended to support the download of other attachment types?

https://github.com/microsoft/teams-ai/blob/913423b84528e30749f03805459546e29b0e292d/js/packages/teams-ai/src/TeamsAttachmentDownloader.ts#L76
The `TeamsAttachmentDownloader` relies on the contentUrl to download files. But it doesn't work for attachment files. 
- Attachment files in personal chats can be conveniently downloaded using the URL provided in `context.activity.attachments[i].content.downloadUrl`. No authentication token is required for this download url.
- The contentUrl of attachment files in group or channel chats leads to a SharePoint webUrl. This SharePoint webUrl cannot be directly use for file downloads.
   

### Reproduction Steps

```bash
1. Upload file in Teams
2. The `teams-ai` SDK throw an error
```

**Comment by Stevenic** [comment]
Yeah I only implemented images because that’s all that OpenAI supports. Adding support for any type of attachment was what I envisioned I just ran out of time.

**Comment by Benjiiim** [comment]
I would be very interested for this feature as well.

**Comment by Benjiiim** [comment]
Hi @corinagum, hi all,
I'm wondering if this enhancement is being considered for the future?
Thanks a lot

**Comment by Benjiiim** [comment]
Hi @corinagum,
I'm wondering if you can share some insights regarding the work needed to receive messages' attachements' content with v2.
Thanks a lot

**Comment by Benjiiim** [comment]
Thanks for your input, @jerbob92.
I’ve postponed the file-related features for my agent as long as I could—I was hoping the Teams SDK would help at some point.
Still hoping for some good news from @corinagum :-)

**Comment by jerbob92** [comment]
Ideally the SDK would provide a downloader for all these different types.
But for the SharePoint URLs it would also be a lot better if there would be some RSC permission for it. Setting up sharepoint site permissions for apps is a real pain.

**Comment by Benjiiim** [comment]
I'm glad this issue survived the great cleanup by @corinagum and team ;-)
